### DISCUSSION TITLE
Timeline for Angular 18 support

### DISCUSSION THREAD

**Discussion by scottwilkin** [stephanrauh/ngx-extended-pdf-viewer#2487] [General]
Is there a tentative schedule for releasing a version of the viewer with support for Angular 18? I was recently working on an update to Angular 18 for an app that includes your module (love it by the way 😄 ) and noticed the dependency conflict:

```
└─┬ ngx-extended-pdf-viewer 19.7.1
  ├── ✕ unmet peer @angular/core@">=16.0.0 <18.0.0": found 18.1.3
  └── ✕ unmet peer @angular/common@">=16.0.0 <18.0.0": found 18.1.3
```

Since the viewer is a key feature of the app in question we don't want to risk it breaking so we're putting the version update on hold for the time being. But for planning purposes I was wondering if there was a date in mind for updating the viewer?

**Comment by stephanrauh** [comment]
Update to version 21.1.x. There seem to be a few minor bugs of the new version (#2485, #2486), but that shouldn't be much of the problem - I'll fix these glitches soon.

Version 20.5.0 was the first version that officially supported Angular 18. However, version 20.5.0 was an intermediate version, so I expect version 21.1.x to be more mature. I suppose you'll also notice a performance boost in version 21.1.x.

To be honest, the older versions also supported it, but for some reason unknown I forgot to document that in the package.json.
